src/parse.jl

```diff
@@ -0,0 +1,33 @@
+const INTERVAL_REGEX = r"""
+    (\[|\()                # Left bound type (1)
```

Comment:
These comments are very helpful :)

Reply:
Do the numbers make sense too? You can make named groups with regex but that doesn't work with alternative captures (`?|`)

Reply:
Yes!